lib/model/model.go

```diff
@@ -2550,24 +2556,39 @@ func makeForgetUpdate(files []protocol.FileInfo) []protocol.FileDownloadProgress
 
 // shouldIgnore returns true when a file should be excluded from processing
 func shouldIgnore(file db.FileIntf, matcher *ignore.Matcher, ignoreDelete bool) bool {
-	// We check things in a certain order here...
-
 	switch {
 	case ignoreDelete && file.IsDeleted():
 		// ignoreDelete first because it's a very cheap test so a win if it
 		// succeeds, and we might in the long run accumulate quite a few
 		// deleted files.
 		return true
 
+	case isInternal(file.FileName()):
+		return true
+
 	case matcher.Match(file.FileName()).IsIgnored():
-		// ignore patterns second because ignoring them is a valid way to
-		// silence warnings about them being invalid and so on.
 		return true
 	}
 
 	return false
 }
 
+func isInternal(name string) bool {
```

Comment:
Yeah. I'll look at the parent trimming thing. That's a somewhat different thing though, as it gets fed via a path that cleans names and should only be looking at things that are already below the specified root.